### PR TITLE
#14 Allow generic token_options to be passed via $config[]

### PR DESCRIPTION
**Pull request by jonathan-graf**
This enhancement will allow for other token options to be passed through the middleware for compatibility with OAuth2 compatible frameworks that have some "extra" requirements.

### CODE REVIEW

**Comment by akira28** [comment]
Do you mind adding a test covering this addition? Thank you very much for your contribution!

**Comment by joskfg** [comment]
In addition to the test, It would be good to add some documentation in the usage part of the README.md

Thanks!

File: src/AddAuthorizationHeader.php [hunk 1]
```diff
@@ -55,6 +55,14 @@ private function getOptions(): array
         if (!empty($this->config['scope'])) {
             $options['scope'] = $this->config['scope'];
         }
+      
+        if (!empty($this->config['token_options'])) {
+            $token_options = $this->config['token_options'];
```
Comment (rccrdpccl) [inline src/AddAuthorizationHeader.php]: Please use camelCase for variable names

**Review by rccrdpccl — CHANGES_REQUESTED** [review]
Thanks for the improvement! This looks good, just a few minor changes and it'll get merged.

**Comment by jonathan-graf** [comment]
> Do you mind adding a test covering this addition? Thank you very much for your contribution!

Not a problem!

**Review by rccrdpccl — COMMENTED** [review]
What happens when we both declare scope in root config and in `token_options`? Could you please explicit what happens covering this case in the tests? IMO the behaviour is correct, but it is a bit of a quirk to keep backward compatibility, so it would be nice to add a test for it. With this small change is good to merge IMO

**Comment by jonathan-graf** [comment]
What what you like the intended result to be?  Should we throw an exception
and force scope to only be in root config?

**Comment by rccrdpccl** [comment]
I think current behaviour it's good: if it's defined in root config, it works. If it's set in `token_options` it also works. If it's set in both, the new `token_options` it's used. This way the library it's backward compatible (current clients won't break). In case it's declared twice it's a weird scenario, that's why I think it should be explicitly tested.

**Comment by jonathan-graf** [comment]
Sounds good.  Thanks for clearing that up for me.  I’ll know which
assertions to code very clearly now.

**Review by rccrdpccl — APPROVED** [review]

**Review by joskfg — APPROVED** [review]

**Review by akira28 — APPROVED** [review]

**Pull request merged**